app/controllers/request-to-go-live/organisation-name/post.controller.js

```diff
@@ -5,27 +5,36 @@ const lodash = require('lodash')
 
 // Local dependencies
 const goLiveStageToNextPagePath = require('../go-live-stage-to-next-page-path')
+const goLiveStage = require('../../../models/go-live-stage')
 const { requestToGoLive } = require('../../../paths')
 const { validateOrganisationName } = require('../../../utils/organisation_name_validation')
-
+const { updateCurrentGoLiveStage, updateMerchantName } = require('../../../services/service_service')
+const { renderErrorView } = require('../../../utils/response')
 // Constants
-const REQUEST_ORGANISATION_NAME_FIELD = 'organisation-name'
+const ORGANISATION_NAME_FIELD = 'organisation-name'
 
 module.exports = (req, res) => {
-  const organisationName = lodash.get(req, 'body.organisation-name')
-  const errors = validateOrganisationName(organisationName, REQUEST_ORGANISATION_NAME_FIELD, true)
-
+  const organisationName = lodash.get(req.body, ORGANISATION_NAME_FIELD)
+  const errors = validateOrganisationName(organisationName, ORGANISATION_NAME_FIELD, true)
   if (lodash.isEmpty(errors)) {
-    // TODO: handle submission
-    res.redirect(
-      303,
-      goLiveStageToNextPagePath[req.service.currentGoLiveStage].replace(':externalServiceId', req.service.externalId)
-    )
+    return updateMerchantName(req.service.externalId, organisationName, req.correlationId)
+      .then(service => {
+        return updateCurrentGoLiveStage(service.externalId, goLiveStage.ENTERED_ORGANISATION_NAME, req.correlationId)
+          .then(updatedService => {
```

Comment:
This promise structure is nested in the same way a callback chain might be, this code should actually work but it might become hard to follow in the future, especially for error handling. 

The goal should be to only ever have a depth of 1 in promise chains, that way the error is handled uniformly at the bottom and there aren't small issues with promises not being returned further down the chain. 

(we also don't need to return anything to express route handlers)

This would look something like: 
```js 
updateMerchantName()
  .then((service) => { 
    return updateCurrentGoLiveStage()
  })
  .then((updatedService) => { 
    // service successfully updated, redirect user to next stage
    res.redirect()
  })
  .catch((error) => { 
    renderErrorView()
  })
```

Reply:
Awesome thanks!  I've updated this now 👌🏿